src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -265,45 +266,38 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
 
             // Update declarations parameter lists
 
-            if (updatedNode.IsKind(SyntaxKind.MethodDeclaration))
+            if (updatedNode.IsKind(SyntaxKind.MethodDeclaration, out MethodDeclarationSyntax method))
```

Comment:
Isn't this better to be switch-case, even with types, because type switches will be optimized at some point. (https://github.com/dotnet/corefx/issues/36132)

Reply:
Perhaps in the future, but we aren't there today.